### PR TITLE
Adapt the KnotInfo interface to new Khovanov polynomial data (merged)

### PR DESCRIPTION
**Pull request by soehms**
# CAUTION

## Although this PR is not marked as *merged* it has been merged into 10.1.beta6

Note, that `Files changed 0`!  For more information see https://github.com/sagemath/sage/pull/35800#issuecomment-1643762556.

### :books: Description

A note in the current [Khovanov polynomial](https://doc.sagemath.org/html/en/reference/knots/sage/knots/knotinfo.html#sage.knots.knotinfo.KnotInfoBase.khovanov_polynomial) method of the KnotInfo interface says:

> The data used here were calculated with the program KhoHo. They are no longer visible on the website as of October 30, 2022. Instead, data computed with KnotJob are now displayed ... This interface will be adapted to the changes in an upcoming release.

The goal of this PR is the announced adaptation. Accordingly the PR lets Sage point to the current version [2023.6.1](https://github.com/soehms/database_knotinfo/releases/tag/2023.6.1) of the [Python wrapper](https://github.com/soehms/database_knotinfo#readme) of the database which contains the new columns.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by soehms** [comment]
I don't know why all GitHub bots except the one building the documentation have been cancelled? Anyway, the changes of the branch do only effect `src/sage/knots/knotinfo.py` and here all tests a fine.

File: src/sage/knots/knotinfo.py [hunk 1]
```diff
@@ -1738,6 +1748,35 @@ def khovanov_polynomial(self, var1='q', var2='t', base_ring=ZZ, original=False):
...
+            True
+            sage: K.khovanov_polynomial(base_ring=QQ, odd=True) == Kkr
+            True
+            sage: K.khovanov_polynomial(base_ring=GF(2), odd=True) == Kkr2
+            True
+            sage: K8ko = K8_19.khovanov_polynomial(odd=True); K8ko     # optional database_knotinfo
+            q^16*t^5 + q^10*t^2 + q^6
+            sage: K8kr == K8ko                                         # optional database_knotinfo
+            False
+
```
Comment (tscrim) [inline src/sage/knots/knotinfo.py]: One too many blank lines. `;)`

File: src/sage/knots/knotinfo.py [hunk 2]
```diff
@@ -1747,8 +1786,65 @@ def khovanov_polynomial(self, var1='q', var2='t', base_ring=ZZ, original=False):
             True
             sage: Lk == L.link().khovanov_polynomial()
             True
+
+        TESTS:
```
Comment (tscrim) [inline src/sage/knots/knotinfo.py]:
```suggestion
        TESTS::
```

File: src/sage/knots/knotinfo.py [hunk 3]
```diff
@@ -1696,8 +1696,16 @@ def khovanov_polynomial(self, var1='q', var2='t', base_ring=ZZ, original=False):
...
+        - ``original`` -- boolean (default: ``False``) if set to
           ``True`` the original table entry is returned as a string
+        - ``reduced`` -- boolean (default: ``False``) if set to ``True``
+          the reduced version of the homology is used.
+        - ``odd`` -- boolean (default: ``False``) if set to ``True``
+          the odd version of the homology is used.
+        - ``KhoHo`` -- boolean (default: ``False`` for knots and ``True``
+          for multi-component links) if set to ``True`` the data calculated
+          using `KhoHo <https://github.com/AShumakovitch/KhoHo>`__ is used
+          (see the note below).
```
Comment (tscrim) [inline src/sage/knots/knotinfo.py]:
```suggestion
        - ``original`` -- boolean (default: ``False``); if set to
          ``True`` the original table entry is returned as a string
        - ``reduced`` -- boolean (default: ``False``); if set to ``True``
          the reduced version of the homology is used
        - ``odd`` -- boolean (default: ``False``); if set to ``True``
          the odd version of the homology is used
        - ``KhoHo`` -- boolean (default: ``False`` for knots and ``True``
          for multi-component links); if set to ``True`` the data calculated
          using `KhoHo <https://github.com/AShumakovitch/KhoHo>`__ is used
          (see the note below)
```

**Review by tscrim — CHANGES_REQUESTED** [review]
Just a few very minor things.

File: src/sage/knots/knotinfo.py (hunk 1 above)
Comment (soehms) [reply src/sage/knots/knotinfo.py]: Amazingly, pycodestyle doesn't recognize this (probably because it's in the doctest).

Comment (soehms) [reply src/sage/knots/knotinfo.py]: I've added a fix in [this commit](https://github.com/soehms/sage/commit/3291f4d1662405303ea95e50db81b59a8a563ec0) but is not shown here.

**Review by tscrim — APPROVED** [review]
Thank you. LGTM.

**Comment by kwankyu** [comment]
@vbraun: strangely this PR is still open.

**Comment by soehms** [comment]
> @vbraun: strangely this PR is still open.

It seems that indeed commit https://github.com/sagemath/sage/pull/35800/commits/3291f4d1662405303ea95e50db81b59a8a563ec0 (which just removes a blank-line) has not been merged. Probably this keeps it open.

**Comment by vbraun** [comment]
Merge conflict

**Comment by soehms** [comment]
> Merge conflict

@vbraun there is some mystery with this PR. It is already merged into `10.1.beta6` but just one commit (https://github.com/sagemath/sage/commit/3291f4d1662405303ea95e50db81b59a8a563ec0) remains off (which just removes a blank line, see https://github.com/sagemath/sage/pull/35800#issuecomment-1631184511 and https://github.com/sagemath/sage/pull/35172#issuecomment-1629348169 at BTW).

Now, if I merge this branch with `develop` and try to push it back, `git` says `Everything up-to-date`. So what can I do?

Note my review comment: https://github.com/sagemath/sage/pull/35800#discussion_r1252595910: With the missing commit there has been something irregular from the beginning: It was not visible on the `PR` (at least for a couple of hours)  after I've pushed it. Maybe this is related to the problem now!

**Pull request closed without merging**